### PR TITLE
Merge ExceptionsHelper into Exceptions/SQLExceptions

### PR DESCRIPTION
**Pull request by mfussenegger**
There were 3 classes for functions dealing with exceptions but we only
need two:

- `Exceptions` which is part of libs/shared - contains the generic
  functions
- `SQLExceptions` which is part of server - contains additions depending
  on server components.

~There is still some duplication left in regards to unwrapping, that will
be a separate task to follow up on.~ The second commit renames `Exceptions.unwrap` to `firstCause` to clarify the difference to `SQLExceptions.unwrap`. See the commit message for details.

### CODE REVIEW

File: server/src/main/java/io/crate/execution/engine/fetch/FetchCollector.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public StreamBucket collect(IntArrayList docIds) {
                     }
                     setNextDocId(readerContext, docId - subReaderContext.docBase);
                 } catch (IOException e) {
-                    Exceptions.rethrowRuntimeException(e);
+                    throw new UncheckedIOException(e);
```
Comment (matriv) [inline server/src/main/java/io/crate/execution/engine/fetch/FetchCollector.java:106]: Is this correct? just double checking.

**Review by matriv — APPROVED** [review]
lgtm, thjx!  just a double checking question.

File: server/src/main/java/io/crate/execution/engine/fetch/FetchCollector.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/execution/engine/fetch/FetchCollector.java:106]: I don't think it breaks anything and `UncheckedIOException` is more informative than wrapping it in `RuntimeException`

Comment (matriv) [reply server/src/main/java/io/crate/execution/engine/fetch/FetchCollector.java:106]: thx

**Pull request merged**